### PR TITLE
Feature/ball handle get ball

### PR DESCRIPTION
**Pull request by ThijsLuttikhuis**
Ball handle takes a moving ball into account

### CODE REVIEW

File: roboteam_ai/src/control/ballHandling/DribbleForwards.cpp [hunk 1]
```diff
@@ -13,9 +13,10 @@ namespace ai {
 namespace control {
 
 
-RobotCommand DribbleForwards::getRobotCommand(const world::Robot::RobotPtr &r, const Vector2 &targetP,
+RobotCommand DribbleForwards::getRobotCommand(world::Robot::RobotPtr r, const Vector2 &targetP,
         const Angle &targetA) {
-    robot = r;
+
+    robot = std::move(r);
```
Comment (rolfvdhulst) [inline roboteam_ai/src/control/ballHandling/DribbleForwards.cpp]: I personally don't really like std::move here, as you are editing the robotPtr object which becomeso a nullptr now, so if dribbleForwards tries to access the robotptr afterwards is null now. Is there any reason why you made this change?

Comment (rolfvdhulst) [reply roboteam_ai/src/control/ballHandling/DribbleForwards.cpp]: also if this is accessing world directly you are erasing robot pointers in world

File: roboteam_ai/src/control/ballHandling/RotateAroundBall.cpp [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#include <utility>
```
Comment (rolfvdhulst) [inline roboteam_ai/src/control/ballHandling/RotateAroundBall.cpp]: not necessary

File: roboteam_ai/src/control/ballHandling/RotateAroundBall.cpp [hunk 3]
```diff
@@ -10,11 +12,10 @@ namespace rtt {
 namespace ai {
 namespace control {
 
-RobotCommand RotateAroundBall::getRobotCommand(const world::Robot::RobotPtr &r, const Vector2 &targetP,
+RobotCommand RotateAroundBall::getRobotCommand(const world::Robot::RobotPtr &robot, const Vector2 &targetP,
```
Comment (rolfvdhulst) [inline roboteam_ai/src/control/ballHandling/RotateAroundBall.cpp]: If you are going to remove const ref from getRobotCommand do it everywhere please.

File: roboteam_ai/src/control/numTrees/NumTreePosControl.h [hunk 4]
```diff
@@ -36,7 +36,7 @@ class NumTreePosControl : public BasicPosControl {
         RobotCommand computeCommand(const Vector2 &exactTargetPos);
 
         // constants
-        const double MAX_CALCULATION_TIME = 20.0;         // Max calculation time in ms
+        static constexpr double MAX_CALCULATION_TIME = 20.0;         // Max calculation time in ms
```
Comment (rolfvdhulst) [inline roboteam_ai/src/control/numTrees/NumTreePosControl.h]: don't we all love CPP

File: roboteam_ai/src/control/ballHandling/RotateAroundBall.cpp (hunk 2 above)
Comment (ThijsLuttikhuis) [reply roboteam_ai/src/control/ballHandling/RotateAroundBall.cpp]: yes, sorry, fixed

File: roboteam_ai/src/control/ballHandling/RotateAroundBall.cpp (hunk 3 above)
Comment (ThijsLuttikhuis) [reply roboteam_ai/src/control/ballHandling/RotateAroundBall.cpp]: fixed

File: roboteam_ai/src/control/ballHandling/DribbleBackwards.cpp [hunk 5]
```diff
@@ -1,3 +1,5 @@
+#include <utility>
```
Comment (rolfvdhulst) [inline roboteam_ai/src/control/ballHandling/DribbleBackwards.cpp]: Don't forget this one

Comment (ThijsLuttikhuis) [reply roboteam_ai/src/control/ballHandling/DribbleBackwards.cpp]: yep

**Pull request merged**